R/module_filter_manager.R

```diff
@@ -252,3 +231,96 @@ filter_manager_module_srv <- function(id, module_fd, slices_global) {
+# For global app, list of length 1 named "global_filters".
+# For module-specific app, list of length one-per-module, named after modules.
+#' @keywords internal
+#' @noRd
+#'
+create_filtered_data_list <- function(filtered_data_list, module_specific) {
+  if (!module_specific) {
+    # Retrieve the first FilteredData from potentially nested list.
+    # List of length one is named "global_filters" because that name is forbidden for a module label.
+    list(global_filters = unlist(filtered_data_list)[[1]])
+  } else {
+    flatten_nested(filtered_data_list)
+  }
+}
+# Flatten potentially nested list of FilteredData objects while maintaining useful names.
+# Simply using `unlist` would result in concatenated names.
+#' @keywords internal
+#' @noRd
+#'
+flatten_nested <- function(x, name = NULL) {
+  if (inherits(x, "FilteredData")) {
+    setNames(list(x), name)
+  } else {
+    unlist(lapply(names(x), function(name) flatten_nested(x[[name]], name)))
+  }
+}
+
+# Create mapping fo filters to modules in matrix form (presented as data.frame).
+# Modules get NAs for filters that cannot be set for them.
+#' @keywords internal
+#' @noRd
+#'
+create_mapping_matrix <- function(filtered_data_list, slices_global) {
+  reactive({
+    state_ids_global <- vapply(slices_global(), `[[`, character(1L), "id")
+    mapping_smooth <- lapply(filtered_data_list, function(x) {
+      state_ids_local <- vapply(x$get_filter_state(), `[[`, character(1L), "id")
+      state_ids_allowed <- vapply(x$get_available_teal_slices()(), `[[`, character(1L), "id")
+      states_active <- state_ids_global %in% state_ids_local
+      ifelse(state_ids_global %in% state_ids_allowed, states_active, NA)
+    })
+
+    as.data.frame(mapping_smooth, row.names = state_ids_global, check.names = FALSE)
+  })
+}
+
+
+
+# setter for global variable ----
+#' @keywords internal
+#' @noRd
+setter_slices_global <- function() {
+  sesh <- getDefaultReactiveDomain()
+  if (is.null(sesh$userData$slices_global)) {
+    filter <- dynGet("filter")
+    sesh$userData$slices_global <- reactiveVal(filter)
+  } else {
+    sesh$userData$slices_global
+  }
+}
+#' @keywords internal
+#' @noRd
+setter_filtered_data_list <- function() {
+  sesh <- getDefaultReactiveDomain()
+  if (is.null(sesh$userData$filtered_data_list)) {
+    filtered_data_list <- dynGet("filtered_data_list")
+    is_module_specific <- dynGet("is_module_specific")
+    sesh$userData$filtered_data_list <- create_filtered_data_list(filtered_data_list, is_module_specific)
+  } else {
+    sesh$userData$filtered_data_list
+  }
+}
+#' @keywords internal
+#' @noRd
+setter_mapping_matrix <- function() {
+  sesh <- getDefaultReactiveDomain()
+  if (is.null(sesh$userData$mapping_matrix)) {
+    filtered_data_list <- dynGet("filtered_data_list")
+    slices_global <- dynGet("slices_global")
+    sesh$userData$mapping_matrix <- create_mapping_matrix(filtered_data_list, slices_global)
+  } else {
+    sesh$userData$mapping_matrix
+  }
```

Comment:
Existence of `GlobalSetter` complicates this as well. Optimally function `setter_mapping_matrix` should be called once and it shouldn't check whether values are set or not.
```suggestion
```